async_upnp_client/server.py

```diff
@@ -1073,14 +1073,15 @@ async def _parse_action_body(
     try:
         _, action_name = soap_action.split("#")
         data = await request.text()
+        data = data.replace('"s:', '" s:').replace('"u:', '" u:')
         root_el: ET.Element = DET.fromstring(data)
         body_el = root_el.find("s:Body", NAMESPACES)
         assert body_el
         rpc_el = body_el[0]
     except Exception as exc:
         raise HTTPBadRequest(reason="InvalidSoap") from exc
 
-    if action_name not in service.actions:
+    if not any([a for a in service.actions if str(a) == action_name]):
```

Comment:
It would be better to normalize action_name instead so we don't have to do a linear search of actions

Reply:
Changed to use only dict keys.